File: conformance/tests/state-ripemd160/state-ripemd160.runar.go
//go:build ignore

package contract

import "runar"

type HashRegistry struct {
	runar.StatefulSmartContract
	CurrentHash runar.Ripemd160
}

func (c *HashRegistry) Update(newHash runar.Ripemd160) {
	c.CurrentHash = newHash
	runar.Assert(true)
}

File: conformance/tests/state-ripemd160/state-ripemd160.runar.ts
import { StatefulSmartContract, assert } from 'runar-lang';

class HashRegistry extends StatefulSmartContract {
  currentHash: Ripemd160;

  constructor(currentHash: Ripemd160) {
    super(currentHash);
    this.currentHash = currentHash;
  }

  public update(newHash: Ripemd160): void {
    this.currentHash = newHash;
    assert(true);
  }
}
